src/main/scala/io/iohk/ethereum/transactions/PendingTransactionsManager.scala

```diff
@@ -99,12 +107,24 @@ class PendingTransactionsManager(txPoolConfig: TxPoolConfig, peerManager: ActorR
     case RemoveTransactions(signedTransactions) =>
       pendingTransactions = pendingTransactions.filterNot(pt => signedTransactions.contains(pt.stx))
       knownTransactions = knownTransactions.filterNot(signedTransactions.map(_.hash).contains)
+      signedTransactions.foreach(clearTimeout)
 
     case MessageFromPeer(SignedTransactions(signedTransactions), peerId) =>
       self ! AddTransactions(signedTransactions.toList)
       signedTransactions.foreach(setTxKnown(_, peerId))
   }
 
+  private def setTimeout(stx: SignedTransaction): Unit = {
+    timeouts.get(stx.hash).map(_.cancel())
+    val cancellable = context.system.scheduler.scheduleOnce(txPoolConfig.transactionTimeout, self, RemoveTransactions(Seq(stx)))
+    timeouts += (stx.hash -> cancellable)
+  }
+
+  private def clearTimeout(stx: SignedTransaction): Unit = {
```

Comment:
The name... I'd say what we do about the `stx` is more essential. Also `clearTimeout` may sound like we reset it. What about `forgetTransaction` or something like that?